query/physicalplan/aggregate.go

```diff
@@ -424,6 +324,13 @@ func (a *HashAggregate) Callback(ctx context.Context, r arrow.Record) error {
 				groupByFields = append(groupByFields, field)
 				groupByArrays = append(groupByArrays, r.Column(i))
 
+				if a.finalStage { // in the final stage expect the hashes to already exist, so only need to combine them as normal hashes
+					groupByFieldHashes = append(groupByFieldHashes,
+						&uint64HashCombine{value: scalar.Hash(a.hashSeed, scalar.NewStringScalar(field.Name))},
```

Comment:
My concern here is that won't the seeds across nodes/restarts be unstable and therefore the hashes won't match? Might be missing something.

Reply:
This seed is just for the hash combiner. Which yes will be inconsistent between restarts, but consistent for any given query so it isn't a problem since it'll combine them in the same way at each stage.

Reply:
Technically the final stage doesn't even need to use the hashcombiner since it could just re-use the hashes from the previous stages. But I didn't see any performance gains when I tried this.

Reply:
However with a distributed query engine it could be problematic to generate different combined hashes on each instance

Reply:
Ahh gotcha. So since only the final stage generates a seed this is not an issue since the distributor will be the only one to do so.

Reply:
We inject a single seed for the whole query. But the final stage drops the combined hashes